### PR TITLE
Use public sign

### PR DESCRIPTION
**Pull request by sharwell**
Use PublicSign instead of DelaySign, for cases where the latter was used in developer builds.

This addresses the assembly strong naming limitations involved in #6032.

### CODE REVIEW

**Comment by cartermp** [comment]
Link #6032

**Comment by TIHan** [comment]
Adding @dsyme  as he needs to see this

File: FSharpBuild.Directory.Build.props [hunk 1]
```diff
@@ -42,7 +42,10 @@
     <SignAssembly>true</SignAssembly>
     <AssemblyOriginatorKeyFile>$(FSharpSourcesRoot)\fsharp\msft.pubkey</AssemblyOriginatorKeyFile>
     <StrongNames>true</StrongNames>
-    <DelaySign>true</DelaySign>
+    <DelaySign>false</DelaySign>
+    <PublicSign>true</PublicSign>
+    <GenerateSatelliteAssembliesForCore>true</GenerateSatelliteAssembliesForCore>
```
Comment (KevinRansom) [inline FSharpBuild.Directory.Build.props]: Why is this necessary?

File: FSharpBuild.Directory.Build.props [hunk 2]
```diff
@@ -42,7 +42,10 @@
...
+    <NoCompilerStandardLib Condition="'$(TargetFramework)' == 'netstandard1.6'">true</NoCompilerStandardLib>
```
Comment (KevinRansom) [inline FSharpBuild.Directory.Build.props]: Ditto?

**Comment by cartermp** [comment]
Don is currently on vacation, so if this isn't controversial (I can't see how it would be) then we should merge it if it works.

File: FSharpBuild.Directory.Build.props (hunk 2 above)
Comment (KevinRansom) [reply FSharpBuild.Directory.Build.props]: Some additional comments.
- What happens with ngen if run in an administrator window?
- Have you run F# in the ide with this change?
- Does FSI run as expected in an F5 environment?

**Comment by sharwell** [comment]
This isn't blocking me; I'd like to get the opinion of at least one expert in MSBuild and/or the SDK targets.

File: FSharpBuild.Directory.Build.props (hunk 1 above)
Comment (KevinRansom) [reply FSharpBuild.Directory.Build.props]: I mean the GenerateSatelliteAssembliesForCore not the public sign thingy.
Why was that necessary and similarly the NoCompilerStandardLib?

Comment (sharwell) [reply FSharpBuild.Directory.Build.props]: The `GenerateSatelliteAssemblies` target does not support public sign.

File: FSharpBuild.Directory.Build.props (hunk 2 above)
Comment (sharwell) [reply FSharpBuild.Directory.Build.props]: > What happens with ngen …

NGEN and shadow copying are two actions that trigger strong name validation. They will fail on public-sign assemblies, but they are also easily avoided actions.

> Have you run F# in the ide with this change?

No.

> Does FSI run as expected in an F5 environment?

I have not tested this.

💭 It's not clear whether disabling strong name validation applies to both public-sign and delay-sign assemblies. If it applies to both, then we could leave that portion in place for this change to have no impact on users who still choose to build with admin permissions.

Comment (sharwell) [reply FSharpBuild.Directory.Build.props]: > Ditto?

I can't figure out where this gets set normally. I was expecting it to get set whenever `NoStdLib` is set, but for some reason it's not.

**Comment by dsyme** [comment]
I don't really understand the ramifications of this.  I certainly agree we need to check that NGEN still runs correctly and that VS starts up.   What are the other problems with using public signing?
